language/src/regent/optimize_index_launches.t

```diff
@@ -1213,21 +1379,31 @@ function optimize_index_launch.stat_for_num(cx, node)
     }
   end
 
-  return ast.typed.stat.IndexLaunchNum {
-    symbol = node.symbol,
-    values = node.values,
-    preamble = body.preamble,
-    call = body.call,
-    reduce_lhs = body.reduce_lhs,
-    reduce_op = body.reduce_op,
-    reduce_task = false,
-    args_provably = body.args_provably,
-    is_constant_time = body.is_constant_time,
-    free_vars = body.free_variables,
-    loop_vars = body.loop_variables,
-    annotations = node.annotations,
-    span = node.span,
+  -- If we reach here then either the self interference test failed,
+  -- or the optimization was successful.
+  local index_launch_ast = ast.typed.stat.IndexLaunchNum {
+      symbol = node.symbol,
+      values = node.values,
+      preamble = body.preamble,
+      call = body.call,
+      reduce_lhs = body.reduce_lhs,
+      reduce_op = body.reduce_op,
+      reduce_task = false,
+      args_provably = body.args_provably,
+      is_constant_time = body.is_constant_time,
+      free_vars = body.free_variables,
+      loop_vars = body.loop_variables,
+      annotations = node.annotations,
+      span = node.span,
```

Comment:
I think we should be able to reduce the indent level here by 2 spaces, and then a large part of this diff will go away.

Reply:
Done.